allocator_common.h

```diff
@@ -175,14 +175,13 @@ typedef struct usage_display {
     uint32_t rotation_types;
 } usage_display_t;
 #define USAGE_BASE_DISPLAY 0x0001
-#define USAGE_BASE_DISPLAY_ROTATION_0           0x00000001
-#define USAGE_BASE_DISPLAY_ROTATION_90          0x00000002
-#define USAGE_BASE_DISPLAY_ROTATION_180         0x00000004
-#define USAGE_BASE_DISPLAY_ROTATION_270         0x00000008
-#define USAGE_BASE_DISPLAY_MIRROR_ROTATION_0    0x00000010
-#define USAGE_BASE_DISPLAY_MIRROR_ROTATION_90   0x00000020
-#define USAGE_BASE_DISPLAY_MIRROR_ROTATION_180  0x00000040
-#define USAGE_BASE_DISPLAY_MIRROR_ROTATION_270  0x00000080
+/* 2 bits to describe rotation */
+#define USAGE_BASE_DISPLAY_ROTATION_0           0x00000000
+#define USAGE_BASE_DISPLAY_ROTATION_90          0x00000001
+#define USAGE_BASE_DISPLAY_ROTATION_180         0x00000002
+#define USAGE_BASE_DISPLAY_ROTATION_270         0x00000003
+/* mirror flag */
+#define USAGE_BASE_DISPLAY_MIRROR               0x00000004
```

Comment:
Hmm, I had assumed users would specify multiple full configurations explicitly using one or more bits in the rotation, rather than build up a single configuration from multiple  bits.  However, it's probably better each "usage" structure name just one configuration.  This looks like an improvement in that regard.